app/controllers/bookings_controller.rb

```diff
@@ -1,37 +1,41 @@
 class BookingsController < ApplicationController
+  before_action :set_booking, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @bookings = Booking.where(user_id: current_user.id)
+  end
+
+  def new
+    @booking = Booking.new
+    authorize @booking
+    @camera = Camera.find(params[:camera_id])
+  end
 
   def create
     @camera = Camera.find(params[:camera_id])
     @booking = Booking.new(booking_params)
+    authorize @booking
     @booking.camera = @camera
     @booking.user = current_user
     # @booking.status = "Pending owner validation"
     if @booking.save
-      redirect_to camera_bookings_path
+      redirect_to dashboard_path
     else
       render 'cameras/show'
     end
   end
 
-  def index
-    @bookings = Booking.where(user_id: current_user.id)
-    @review = Review.new
-  end
-
   def show
-    set_booking
     @camera = @booking.camera
   end
 
   def update
-    # set_booking
     # @booking.status = "Pending owner validation"
-    # booking.save!
-    # redirect_to booking_path(@booking)
+    booking.save!
+    redirect_to booking_path(@booking)
```

Comment:
maybe it should be camera_booking_path?

Reply:
Yes you're right. Now that we have a dashboard we'll redirect there instead.